server/app/durablejobs/DurableJobRegistry.java

```diff
@@ -72,30 +75,57 @@ public void registerWithNoTimeResolver(
       DurableJobName jobName, JobType jobType, DurableJobFactory durableJobFactory) {
     validateJobName(jobName);
 
+    registeredJobs.put(
+        jobName.getJobNameString(),
+        RegisteredJob.create(
+            durableJobFactory, jobName, jobType, /* jobExecutionTimeResolver= */ Optional.empty()));
+  }
+
+  /** Registers a factory for a given job name. Cannot be used with {$JobType.RECURRING}. */
+  public void registerStartupJob(
+      DurableJobName jobName, JobType jobType, DurableJobFactory durableJobFactory) {
+    validateJobName(jobName);
+    if (jobType == JobType.RECURRING) {
```

Comment:
**suggestion:** Can we use the allowedJobTypes instead here and below? That way we don't need to remember to update this if there becomes more than one recurring job type?

Reply:
Not cleanly without a bunch of refactoring. You try to pass in the scheduler you'll get a circular reference. Passing in just the list of allowed types would let the caller pass anything, at which point why ask for allowed types if you can get around the safety check.

Reply:
Ah yeah makes sense - I wonder if its worth a comment above the allowedJobTypes for the recurring class that we'll need to update this if that changes?